google-cloud-storage/src/main/java/com/google/cloud/storage/testing/ApiPolicyMatcher.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.testing;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.easymock.EasyMock;
+import org.easymock.IArgumentMatcher;
+
+import com.google.api.services.storage.model.Policy;
+import com.google.api.services.storage.model.Policy.Bindings;
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * Matches two {@link Policy} instances, which may have lists of {@link Bindings} that are not in
+ * the same order but which are still logically equivalent.
+ */
+public class ApiPolicyMatcher implements IArgumentMatcher {
```

Comment:
It is necessary for this to be in `src/main`? If we put it in `src/test` then we could avoid needing to update the dependency.